app/models/graph.rb

```diff
@@ -1,5 +1,13 @@
+require 'timeout'
+
 module Graph
   class << self
+    def query(query_string, variables: {}, context: {}, timeout: nil)
+      Timeout.timeout(timeout) do
```

Comment:
Also it seems like for web requests in the worse case unicorn will kill the process. I guess I probably wouldn't use it in library code though 🤔